internal/npm_install/generate_build_file.js

```diff
@@ -291,44 +299,57 @@ def _maybe(repo_rule, name, **kwargs):
         repo_rule(name = name, **kwargs)
 `;
 
-  // Copy all files for this workspace to a folder under _workspaces
-  // to preserve the Bazel files which will be deleted from the npm package
-  // by deleteBazelFiles()
-  const workspaceSourcePath = path.posix.join('_workspaces', bwName);
-  mkdirp(workspaceSourcePath);
-  bwDetails.pkg._files.forEach(file => {
-    if (/^node_modules[/\\]/.test(file)) {
-      // don't copy over nested node_modules
-      return;
-    }
-    const src = path.posix.join('node_modules', bwDetails.pkg._dir, file);
-    const dest = path.posix.join(workspaceSourcePath, file);
-    mkdirp(path.dirname(dest));
-    console.error(`copying ${src} -> ${dest}`);
-    fs.copyFileSync(src, dest);
-  });
+  if (bwDetails.archive) {
+    bzlFile +=
+        `load("@build_bazel_rules_nodejs//tools:extract_repository.bzl", "extract_repository")
+def install_${bwName}():
+    _maybe(
+        extract_repository,
+        name = "${bwName}",
+        archive = "@${WORKSPACE}//node_modules/${bwDetails.pkg._dir}:${bwDetails.archive}",
+        strip_prefix = "${bwDetails.stripPrefix ? bwDetails.stripPrefix : ''}",
+    )
+`;
+  } else {
+    // Copy all files for this workspace to a folder under _workspaces
+    // to preserve the Bazel files which will be deleted from the npm package
+    // by deleteBazelFiles()
+    const workspaceSourcePath = path.posix.join('_workspaces', bwName);
+    mkdirp(workspaceSourcePath);
+    bwDetails.pkg._files.forEach(file => {
+      if (/^node_modules[/\\]/.test(file)) {
+        // don't copy over nested node_modules
+        return;
+      }
+      const src = path.posix.join('node_modules', bwDetails.pkg._dir, file);
+      const dest = path.posix.join(workspaceSourcePath, file);
+      mkdirp(path.dirname(dest));
+      console.error(`copying ${src} -> ${dest}`);
```

Comment:
is this meant for debugging?

Reply:
Ahh. Yes. Thanks for the catch. This PR is parked for now any but will clean that up.